File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/io/RedirectConsoleOutputStream.java
package org.javamaster.invocationlab.admin.io;


import lombok.extern.slf4j.Slf4j;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

/**
 * @author yudong
 */
@Slf4j
public class RedirectConsoleOutputStream extends PrintStream {
    private final ByteArrayOutputStream byteArrayOutputStream;

    private final PrintStream out = System.out;

    public RedirectConsoleOutputStream(ByteArrayOutputStream byteArrayOutputStream) {
        super(byteArrayOutputStream);
        this.byteArrayOutputStream = byteArrayOutputStream;
    }

    @Override
    public void println(String x) {
        out.println(x);
        super.println(x);
    }

    public byte[] getLogByteArray() {
        return byteArrayOutputStream.toByteArray();
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/io/RedirectConsoleOutputStream.kt
package org.javamaster.invocationlab.admin.io

import java.io.ByteArrayOutputStream
import java.io.PrintStream


/**
 * @author yudong
 */
class RedirectConsoleOutputStream(private val byteArrayOutputStream: ByteArrayOutputStream) : PrintStream(
    byteArrayOutputStream
) {
    private val printStream: PrintStream = System.out

    override fun println(x: Any?) {
        printStream.println(x)
        super.println(x)
    }

    val logByteArray: ByteArray
        get() = byteArrayOutputStream.toByteArray()
}
